Controllers/TimeWorkedAndVacationController.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using stunning_robot_HR.Data;
+using stunning_robot_HR.Models;
+
+namespace stunning_robot_HR.Controllers
+{
+    public class TimeWorkedAndVacationController : Controller
+    {
+        private readonly stunning_robot_HRContext _context;
+
+        public TimeWorkedAndVacationController(stunning_robot_HRContext context)
+        {
+            _context = context;
+        }
+
+        // GET: TimeWorkedAndVacation
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.TimeWorkedAndVacation.ToListAsync());
+        }
+
+        // GET: TimeWorkedAndVacation/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var timeWorkedAndVacation = await _context.TimeWorkedAndVacation
+                .FirstOrDefaultAsync(m => m.TimeWorkedAndVacationId == id);
+            if (timeWorkedAndVacation == null)
+            {
+                return NotFound();
+            }
+
+            return View(timeWorkedAndVacation);
+        }
+
+        // GET: TimeWorkedAndVacation/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+
+        [HttpPost] 
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("TimeWorkedAndVacationId,TotalNumberOfDaysWorked,TotalNumberOfAvailableVacationDays")] TimeWorkedAndVacation timeWorkedAndVacation)
+        {
+            if (ModelState.IsValid)
+            {
+                timeWorkedAndVacation.totalNumberOfAvailableVacationDays = timeWorkedAndVacation.totalNumberOfDaysWorked * 1;
```

Comment:
What I was suggesting here Erin, is that patterns have a purpose. Repository patterns are traditionally this encapuslation of data access logic into a core set of classes called repositories. All of the calls to the database are hidden behind these classes. So traditionally i look for lambda's or linq statements that access the database in controllers. If I see that, I suggest you hide them behind the walls of a repository. 

Additionally, you're doing a bit of a calculation on line 59. I would traditionally call this business logic. It's traditionally the most fun, puzzle like logic to write. There are a pile of different architectural and design approaches to managing this. Yet in our case let's focus on SRP. Single responsibility pattern suggests that each class has one job, and does that job well. Putting this in a repository is a little like a square peg in a round hole. I would literally just create a new layer or folder within my stack called business logic. Within that class I would hide this logic that calculates vacation. Giving that business class one thing to do, manage accrual calculations.